test/e2e/kubectl.go

```diff
@@ -125,7 +125,7 @@ var _ = framework.KubeDescribe("Kubectl client", func() {
 	framework.KubeDescribe("Update Demo", func() {
 		var updateDemoRoot, nautilusPath, kittenPath string
 		BeforeEach(func() {
-			updateDemoRoot = filepath.Join(framework.TestContext.RepoRoot, "docs/user-guide/update-demo")
+			updateDemoRoot = filepath.Join(framework.TestContext.GeneratedAssetsDir, framework.GENERATED_ASSETS, "docs/user-guide/update-demo")
```

Comment:
I am not a fan of this --generated-assests flag. 

Why do we want to support swapping out the assets used for tests? It sounds like a support nightmare. Everything required to build and run the tests should be checked into this depot, and it should not be configurable. 

Isn't the goal something like http://blog.ralch.com/tutorial/golang-embedded-resources/ such that a test can call Asset('testdata/path/to/whatever.yaml') and be assured that this yaml file exists and is compiled into the binary (as opposed to referencing a file on the local disk somewhere?)